runconfig/hostconfig.go

```diff
@@ -217,6 +218,17 @@ func MergeConfigs(config *Config, hostConfig *HostConfig) *ContainerConfigWrappe
 	}
 }
 
+func ValidateHostConfig(hostConfig *HostConfig) error {
```

Comment:
There is already some verification done in daemon, this should all be together...
I'm ok to get this in for 1.7.1 but we need to look at how to better surface validations (and make sure they happen).